src/Pages/KeyboardMapping.js

```diff
@@ -0,0 +1,199 @@
+	{ label: "Comma", value: 0x36 },
+	{ label: "Control Left", value: 0xe0 },
+	{ label: "Delete", value: 0x4c },
+	{ label: "0", value: 0x27 },
+	{ label: "1", value: 0x1e },
+	{ label: "2", value: 0x1f },
+	{ label: "3", value: 0x20 },
+	{ label: "4", value: 0x21 },
+	{ label: "5", value: 0x22 },
+	{ label: "6", value: 0x23 },
+	{ label: "7", value: 0x24 },
+	{ label: "8", value: 0x25 },
+	{ label: "9", value: 0x26 },
+	{ label: "End", value: 0x4d },
+	{ label: "Enter", value: 0x28 },
+	{ label: "Equal", value: 0x2e },
+	{ label: "Escape", value: 0x29 },
+	{ label: "F1", value: 0x3a },
+	{ label: "F2", value: 0x3b },
+	{ label: "F3", value: 0x3c },
+	{ label: "F4", value: 0x3d },
+	{ label: "F5", value: 0x3e },
+	{ label: "F6", value: 0x3f },
+	{ label: "F7", value: 0x40 },
+	{ label: "F8", value: 0x41 },
+	{ label: "F9", value: 0x42 },
+	{ label: "F10", value: 0x43 },
+	{ label: "F11", value: 0x44 },
+	{ label: "F12", value: 0x45 },
+	{ label: "Home", value: 0x4a },
+	{ label: "Intl Backslash", value: 0x31 },
+	{ label: "A", value: 0x04 },
+	{ label: "B", value: 0x05 },
+	{ label: "C", value: 0x06 },
+	{ label: "D", value: 0x07 },
+	{ label: "E", value: 0x08 },
+	{ label: "F", value: 0x09 },
+	{ label: "G", value: 0x0a },
+	{ label: "H", value: 0x0b },
+	{ label: "I", value: 0x0c },
+	{ label: "J", value: 0x0d },
+	{ label: "K", value: 0x0e },
+	{ label: "L", value: 0x0f },
+	{ label: "M", value: 0x10 },
+	{ label: "N", value: 0x11 },
+	{ label: "O", value: 0x12 },
+	{ label: "P", value: 0x13 },
+	{ label: "Q", value: 0x14 },
+	{ label: "R", value: 0x15 },
+	{ label: "S", value: 0x16 },
+	{ label: "T", value: 0x17 },
+	{ label: "U", value: 0x18 },
+	{ label: "V", value: 0x19 },
+	{ label: "W", value: 0x1a },
+	{ label: "X", value: 0x1b },
+	{ label: "Y", value: 0x1c },
+	{ label: "Z", value: 0x1d },
+	{ label: "Meta Left", value: 0xe3 },
+	{ label: "Meta Right", value: 0xe7 },
+	{ label: "Minus", value: 0x2d },
+	{ label: "Numpad Enter", value: 0x58 },
+	{ label: "Page Down", value: 0x4e },
+	{ label: "Page Up", value: 0x4b },
+	{ label: "Period", value: 0x37 },
+	{ label: "Quote", value: 0x34 },
+	{ label: "Semicolon", value: 0x33 },
+	{ label: "Shift Left", value: 0xe1 },
+	{ label: "Shift Right", value: 0xe5 },
+	{ label: "Slash", value: 0x38 },
+	{ label: "Space", value: 0x2c },
+	{ label: "Tab", value: 0x2b }
+];
+
+export default function KeyboardMappingPage() {
+	const { buttonLabels } = useContext(AppContext);
+	const [validated, setValidated] = useState(false);
+	const [saveMessage, setSaveMessage] = useState('');
+	const [keyMappings, setKeyMappings] = useState(baseButtonMappings);
+	const [selectedController] = useState(process.env.REACT_APP_GP2040_CONTROLLER);
+	const [selectedBoard] = useState(process.env.REACT_APP_GP2040_BOARD);
+
+	useEffect(() => {
+		async function fetchData() {
+			setKeyMappings(await WebApi.getKeyMappings());
+		}
+
+		fetchData();
+	}, [setKeyMappings, selectedController]);
```

Comment:
i think process.env needs restart of the bundler to take effect. If that is the case you can remove the hooks.
And just add `const currentBoard = boards[process.env.REACT_APP_GP2040_BOARD];` outside of the component.

The dependency array of the useEffect is only need on mount right?
Then this would suffice:

```javascript
 useEffect(() => {
    WebApi.getKeyMappings().then(setKeyMappings);
  }, []);
```

Reply:
Hi @Pelsin , thanks for checking it out! This looks like it could probably be applied to other Web config inputs as well, would you mind making an Issue for it and writing up a code sample?

You are also welcome to submit a PR and we can take a look, I'm all for cleaning up our React code!